### ISSUE TITLE
SC2144: Don't warn for ksh

### ISSUE THREAD

**Issue by mhoes** [koalaman/shellcheck#1452]
#### For bugs
- Rule Id: SC2144
- [ X] I tried on shellcheck.net and verified that this is still a problem on the latest commit

```sh

#!/bin/ksh

if [ -f /tmp/ls* ]
then
        echo MATCH
else
        echo NO MATCH
fi

```

#### Here's what shellcheck currently says:
SC2144: -e doesn't work with globs. Use a for loop.

#### Here's what I wanted or expected to see:
Although this does indeed fail when the glob matches more than 1 file for most shells I checked (bash, zsh, dash, mksh), this actually works as expected when using the Korn shell:

> touch /tmp/ls-1 /tmp/ls-2
> $ ls /tmp/ls*
> /tmp/ls-1  /tmp/ls-2
> 
> $ ksh ./foo.sh
> MATCH
>

**Comment by koalaman** [comment]
Wow, you're right! This behavior is fascinating:

```
[ -n foo -a ]                # ksh: True,  bash: Error
[ -n foo -invalid bar ]      # ksh: True,  bash: Error
[ -n foo -a -invalid bar ]   # ksh: Error, bash: Error
[ -n foo -a -z bar ]         # ksh: False, bash: False
```

It appears that `ksh` stops processing on any unrecognized tokens, which is what allows `[ -f /tmp/ls* ]` to work, while also having the weird effects above

**Comment by koalaman** [comment]
This is now sort-of fixed. You will instead get a [SC2245](https://github.com/koalaman/shellcheck/wiki/SC2245) explaining that only the first result will be considered, and to use a loop depending on whether you care that *any* or *all* results are regular files.

You can now also use `[ -e *glob* ]` without a warning, since all files produced by a glob match should necessarily exist.
